examples/camera-controller/BUILD.gn

```diff
@@ -122,9 +122,22 @@ executable("chip-camera-controller") {
     "${chip_root}/third_party/libdatachannel:libdatachannel",
   ]
 
-  libs = [ "datachannel" ]
-  lib_dirs =
-      [ rebase_path("${chip_root}/third_party/libdatachannel/repo/build") ]
+  libs = [
+    "datachannel",
+    "usrsctp",
+    "srtp2",
+    "juice",
+  ]
+
+  lib_dirs = [
```

Comment:
needs comments on why we did this: it seems like libdatachannel is not a apt package for ubuntu so we cannot make it part of our build system and require a separate build.

however beyond that, let's add a gni file into third_party/libdatachannel and have constants for these. We should let libdatachannel define `libs` and `dirs` rather than hard-copding this inside separate applications/pkaces. the apps will only have `lib_dirs = datachannel_dirs` and `libs = `datachannel_libs`

Reply:
I would also argue in favor of not including the inner dependencies inside the top level BUILD.gn and instead have a BUILD.gn in libdatachannel that produces the static library using all its internal dependencies. Maybe a followup issue.
Do we have any precedence of a third party library linked as a shared lib? If so, how do they get built in CI?

Reply:
The final goal is the switch to BUILD.gn, but we need to switch all transitive dependencies (libsrtp, libjuice, usrsctp, etc.) also to BUILD.gn,  it is not a trivial task.